tests/components/upcloud/test_config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Tests for the UpCloud config flow."""
+
+import requests.exceptions
+from requests_mock import ANY
+from upcloud_api import UpCloudAPIError
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.upcloud.const import DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+FIXTURE_USER_INPUT = {
+    CONF_USERNAME: "user",
+    CONF_PASSWORD: "pass",
+}
+
+FIXTURE_USER_INPUT_OPTIONS = {
+    CONF_SCAN_INTERVAL: "120",
+}
+
+
+async def test_show_set_form(hass):
+    """Test that the setup form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=None
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_connection_error(hass, requests_mock):
+    """Test we show user form on connection error."""
+    requests_mock.request(ANY, ANY, exc=requests.exceptions.ConnectionError())
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=FIXTURE_USER_INPUT
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "cannot_connect"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)